src/selfplay/tournament.cc

```diff
@@ -166,37 +161,26 @@ SelfPlayTournament::SelfPlayTournament(const OptionsDict& options,
   }
 }
 
-void SelfPlayTournament::PlayOneGame(int game_number) {
-  bool player1_black;  // Whether player1 will player as black in this game.
-  {
-    Mutex::Lock lock(mutex_);
-    player1_black = next_game_black_;
-    next_game_black_ = !next_game_black_;
-  }
+// Creates a fresh SelfPlayGame.
+std::unique_ptr<SelfPlayGame> SelfPlayTournament::CreateNewGame(
+    int game_number) {
+  const bool player1_black = (game_number % 2) != first_game_is_flipped_;
   const int color_idx[2] = {player1_black ? 1 : 0, player1_black ? 0 : 1};
 
   PlayerOptions options[2];
 
-  std::vector<ThinkingInfo> last_thinking_info;
   for (int pl_idx : {0, 1}) {
     const bool verbose_thinking =
         player_options_[pl_idx].Get<bool>(kVerboseThinkingId.GetId());
     // Populate per-player options.
     PlayerOptions& opt = options[color_idx[pl_idx]];
-    opt.network = networks_[pl_idx].get();
     opt.cache = cache_[pl_idx].get();
     opt.uci_options = &player_options_[pl_idx];
     opt.search_limits = search_limits_[pl_idx];
 
     // "bestmove" callback.
-    opt.best_move_callback = [this, game_number, pl_idx, player1_black,
-                              verbose_thinking,
-                              &last_thinking_info](const BestMoveInfo& info) {
-      // In non-verbose mode, output the last "info" message.
-      if (!verbose_thinking && !last_thinking_info.empty()) {
```

Comment:
Yeah it PR was kind of large enough so I thought I'd reintroduce it later.
But I can do that in that PR too if you think it worth it.

(for me it doesn't seem important because
- We use selfplay for training mostly anyway
- It doesn't slow anything down
- It doesn't take that much memory
Yeah it's kind of a waste to build it in memory if we are not going to dump it but I think it's fine. :)
)